src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,426 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+Sometimes, specified scripts has to be executed before and/or after a particular task. To do that, the solution provides
+ `pre_script` and `post_script` pragmas.
+
+To add a pre-script to a task, please use:
+
+```python
+#%pre_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./PRE_SCRIPT_FILE.py])
+```
+
+To add a post-script to a task, use:
+
+```python
+#%post_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./POST_SCRIPT_FILE.py])
+```
+
+==== Create a job
+
+To create a job, use the `#%job()` pragma:
+
+```python
+#%job(name=JOB_NAME)
+```
+
+If the job has already been created, the call of this pragma would just rename the job already created by the new provided name.
+
+**Notice that it is not necessary to create and name explicitly the job. If not done by the user, this step is implicitly
+performed when the job is submitted (check section <<Submit your job to the scheduler>> for more information).**
+
+==== Visualize job
+
+To visualize the created workflow, use the `#%draw_job()` pragma to plot the graph that represents the job into a separate window:
+
+```python
+#%draw_job()
+```
+
+Two optional parameters can be used to configure the way the kernel plots the workflow graph.
+
+**inline plotting:**
+
+If this parameter is set to 'off', the workflow plotting is made through the https://matplotlib.org/[Matplotlib]
+external window. The default value is 'on'.
+
+```python
+#%draw_job(inline=off)
+```
+
+**saving into hard disk:**
+
+To be sure the workflow is saved into a _.png_ file, this option needs to be set to *on*. The default value is *off*.
+
+```python
+#%draw_job(save=on)
+```
+
+Note that the job's name can take one of the following possible values:
+
+1. The parameter `name`'s value, if provided
+2. The job's name, if created
+3. The notebook's name, if the kernel can retrieve it
+4. `Unnamed_job`
+
+General usage:
+
+```python
+#%draw_job([name=JOB_NAME], [inline=off], [save=on])
+```
+
+==== Save workflow in dot format
+
+To save the created workflow into a https://www.graphviz.org/[GraphViz] _.dot_ format, use the `#%write_dot()` pragma:
+
+```python
+#%write_dot(name=FILE_NAME)
+```
+
+
+==== Submit your job to the scheduler
+
+To submit the job to the ProActive Scheduler, the user has to use the `#%submit_job()` pragma:
+
+```python
+#%submit_job()
+```
+
+If the job is not created, or is not up-to-date, the `#%submit_job()` starts by creating a new job named as the old one.
+To provide a new name, use the same pragma and provide a name as parameter:
+
+```python
+#%submit_job(name=JOB_NAME)
+```
+
+Isf the job's name is not set, the ProActive kernel uses the current notebook name, if possible, or gives a random one.
+
+==== List all submitted jobs
+
+To get all the submitted job ids and names, use `list_submitted_jobs` pragma this way:
+
+```python
+#%list_submitted_jobs()
+```
+
+==== Printing results
+
+To get the job result(s), the user has to use the `#%get_result()` pragma by providing the job name:
```

Comment:
provide a reference to  job result(s) doc provided by gleb few weeks ago

Reply:
It is not the same, yet. This feature depends on the Proactive Python Client and still has to be improved. This is why I don't give more information about it. It will have to be changed anyway soon.
(We just print the value of the result ProActive variable if set, or true if the task succeeded.)

Reply:
ok